sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala

```diff
@@ -74,25 +74,30 @@ object ColumnarPartitionReaderWithPartitionValues extends Arm {
       fileBatch: ColumnarBatch,
       partitionValues: Array[Scalar],
       sparkTypes: Array[DataType]): ColumnarBatch = {
-    var partitionColumns: Array[GpuColumnVector] = null
-    try {
-      partitionColumns = buildPartitionColumns(fileBatch.numRows, partitionValues, sparkTypes)
-      val fileBatchCols = (0 until fileBatch.numCols).map(fileBatch.column)
-      val resultCols = fileBatchCols ++ partitionColumns
-      val result = new ColumnarBatch(resultCols.toArray, fileBatch.numRows)
-      fileBatchCols.foreach(_.asInstanceOf[GpuColumnVector].incRefCount())
-      partitionColumns = null
-      result
-    } finally {
-      if (fileBatch != null) {
-        fileBatch.close()
-      }
-      if (partitionColumns != null) {
-        partitionColumns.safeClose()
+    if (fileBatch != null ) {
```

Comment:
Nit: Is this null check necessary?  The original code evaluated `fileBatch.numRows` without checking.

Reply:
I don't think so, the original code had that check in the finally block so I left it. The only place I could see it possibly being null is when its called from above:

 ```
val fileBatch: ColumnarBatch = fileReader.get()
      ColumnarPartitionReaderWithPartitionValues.addPartitionValues(fileBatch,
        partitionValues, partValueTypes)
```

and the fileReader.get() would return null.  I'm fine with removing it, perhaps code changed along the way and that check was never cleaned up?

Reply:
> the original code had that check in the finally block so I left it.

That protects code in the `finally` block, but before we get there the code would have blindly dereferenced `fileBatch` first.  That means we wouldn't get an NPE while trying to cleanup from throwing an NPE. :smile:

I'm totally cool with leaving it in if we think there's a remote chance a `get()` could return null which is why I went ahead and approved it.  However I would expect `next()` to return false before `get()` is allowed to return `null`.